plugin/cmd/kube-scheduler/app/server.go

```diff
@@ -114,7 +114,7 @@ func Run(s *options.SchedulerServer) error {
 	rl := &resourcelock.EndpointsLock{
 		EndpointsMeta: metav1.ObjectMeta{
 			Namespace: "kube-system",
-			Name:      "kube-scheduler",
+			Name:      s.SchedulerName,
```

Comment:
@liggitt Do you have any good  ideas? I almost exhausted on this issue, change the default scheduler name or change the lock location all have compatible problem as you said, 
how about this:
Don't change the scheduler name, don't change the default lock endpint If it's default scheduler, and if it's custom scheduler, we can create a new lock endpoint, but we need fig out what the lock object name should be.